core/sail/base/src/main/java/org/eclipse/rdf4j/sail/base/SailDatasetImpl.java

```diff
@@ -267,15 +267,16 @@ public void handleClose() throws SailException {
 		}
 		Model deprecated = changes.getDeprecated();
 		if (deprecated != null && iter != null) {
-			iter = difference(iter, deprecated.filter(subj, pred, obj, contexts));
+			iter = difference(iter, deprecated);
```

Comment:
I don't think this change makes a difference for the results. The set of statements that are in `iter` but not in `deprecated` is the same unless the pattern used to filter `deprecated` is different from the pattern used by `iter`.

The difference is for performance. Now later on in the FilterIteration we can work directly with the hashmap without having to upgrade to a full LinkedHashModel because of a complicated `filter` operation.

Reply:
The test suite appears to agree with you!